### PR TITLE
Initial functional streams support

### PR DESCRIPTION
**Pull request by tonyvelichko**
Add fs2 support in cases:
* stream from reader
* simple stream for subscription
* batch stream for subscription
* committable sink
* plain sink

### CODE REVIEW

**Comment by tonyvelichko** [comment]
@sksamuel @gmethvin guys have a look 🙏

**Comment by tonyvelichko** [comment]
@gmethvin @sksamuel Guys?

File: pulsar4s-fs2/src/main/scala/com/sksamuel/pulsar4s/fs2/Streams.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+trait CommittableMessage[F[_], X] {
+  def ack: F[Unit]
+  def nack: F[Unit]
+  def data: X
+
+  def map[Y](f: X => Y): CommittableMessage[F, Y]
+}
+
+object Streams {
```
Comment (gmethvin) [inline pulsar4s-fs2/src/main/scala/com/sksamuel/pulsar4s/fs2/Streams.scala:16]: Maybe you want a less generic name here? `PulsarStreams`?

**Review by gmethvin — COMMENTED** [review]
This looks ok. I had a minor suggestion about the naming of `Streams`.

Otherwise I think it's mergeable.

**Comment by sksamuel** [comment]
I will merge this in and rename Streams to PulsarStreams

**Pull request merged**